QuantConnect.BybitBrokerage/Api/BybitHistoryApi.cs

```diff
@@ -0,0 +1,171 @@
+    /// </summary>
+    /// <param name="category">The product category</param>
+    /// <param name="ticker">The ticker to fetch the history for</param>
+    /// <param name="date">The requested date</param>
+    /// <returns>An IEnumerable containing the ticks from the requested date</returns>
+    private IEnumerable<BybitTickUpdate> Download(string ticker, BybitProductCategory category, DateTime date)
+    {
+        if (category is not (BybitProductCategory.Inverse or BybitProductCategory.Linear or BybitProductCategory.Spot))
+        {
+            throw new NotSupportedException("Only inverse, linear, and spot supported");
+        }
+
+        var categoryPath = category == BybitProductCategory.Spot ? "spot" : "trading";
+        var dateSeparator = category == BybitProductCategory.Spot ? "_" : string.Empty;
+        var endpoint = $"/{categoryPath}/{ticker}/{ticker}{dateSeparator}{date:yyyy-MM-dd}.csv.gz";
+
+
+        using var memoryStream = new MemoryStream();
+        DownloadArchiveToStream(endpoint, memoryStream);
+        memoryStream.Position = 0;
+
+        using var gzip = new GZipStream(memoryStream, CompressionMode.Decompress);
+        using var streamReader = new StreamReader(gzip);
+
+        streamReader.ReadLine(); //header
+        var line = streamReader.ReadLine();
+        while (!string.IsNullOrEmpty(line))
+        {
+            if (category == BybitProductCategory.Spot)
+            {
+                yield return ParseSpotTick(line, ticker);
+            }
+            else if (category is BybitProductCategory.Inverse or BybitProductCategory.Linear)
+            {
+                var tick = ParseFuturesTick(line);
+                if (tick != null) yield return tick;
+            }
+            else
+            {
+                throw new NotSupportedException();
+            }
+
+            line = streamReader.ReadLine();
+        }
+    }
+
+    private void DownloadArchiveToStream(string endpoint, Stream stream)
+    {
+        var req = new RestRequest(endpoint)
+        {
+            ResponseWriter = responseStream => responseStream.CopyTo(stream)
+        };
+
+        var response = _restClient.Execute(req);
+
+        if (!response.IsSuccessful)
+        {
+            throw new Exception("ByBitApiClient request failed: " +
+                                $"[{(int)response.StatusCode}] {response.StatusDescription}, " +
+                                $"Content: {response.Content}, ErrorMessage: {response.ErrorMessage}");
+        }
+    }
+
+    private static BybitTickUpdate ParseSpotTick(string line, string ticker)
+    {
+        var tick = new BybitTickUpdate();
+        var split = line.Split(',');
+        tick.Time = DateTimeOffset.FromUnixTimeMilliseconds(long.Parse(split[1])).UtcDateTime;
+        tick.Price = decimal.Parse(split[2], CultureInfo.InvariantCulture);
+        tick.Value = decimal.Parse(split[3], CultureInfo.InvariantCulture);
```

Comment:
We were talking about this during the review, in the case of `BybitTickUpdate` `Value` equals leans quantity. So this is fine.

Reply:
If `.Value` is the traded quantity/volume let's please rename it to Quantity to match the name in the lean Tick class, to avoid confusion because in Lean Value is Price (see https://github.com/QuantConnect/Lean/blob/master/Common/Data/BaseData.cs#L125)